File: labs/06/Average.java
/* Average.java is a driver for method avg().
 *
 * Started by: Prof. Adams, for CS 214 at Calvin College
 *
 * Student Name: Bryce Allen
 * Date: 3/14/18
 *
 * Precondition: theArray, is an array of double values.
 * Output: the average of the numbers in theArray.
 ***************************************************************/

import java.io.*;

public class Average
{
  public static void main(String[] args) {

      double[] array1 = {9.0, 8.0, 7.0, 6.0};
      double[] array0 = new double[0];

      // output the average of the values in the arrays
      System.out.println("The first average is " + avg(array0));
      System.out.println("The second average is " + avg(array1));
  }
  

  /* sum() is a method that computes the sum of all the values
   * in an array of doubles.
   * Receive: anArray, an array of doubles.
   * Return: the sum of the values in anArray.
   ************************************************************/

    public static double sum(double[] anArray)
    {
	double total = 0.0;
	for(int i = 0; i<anArray.length; i++)
	    { total += anArray[i]; }
	return total;
    }
  

  /* avg() is a method that computes the average of the values
   * in an array.
   * Receive: anArray, an array of doubles.
   * Return: the average of the values in anArray.
   **************************************************************/
  
    public static double avg(double[] theArray)
    {
	if(theArray == null || theArray.length <= 0)
	    return 0.0;
	else
	    return sum(theArray)/theArray.length;
    }

}

File: labs/06/average.rb
# average.rb averages the values of an array of doubles.
# Precondition: theArray is an array of numbers
# Output: the average of the values in theArray
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Bryce Allen
# Date:   3/15/18
########################################################

###############################################
# sum() sums the values in an array
# Receive: anArray, an array of numbers
# Return: the sum of the values in anArray.
################################################

def sum(anArray)
    sum = 0.0
   # for i in 0...anArray.size
    #  sum = sum + anArray[i]
   # end
    anArray.each{|i| sum = sum+i}
    return sum
  end
  

###############################################
# average() averages the values in an array
# Receive: anArray, an array of numbers
# Return: average of the values in the array
################################################

def average(anArray)
  return sum(anArray)/anArray.size
end

def main
  array0 = Array.new
  array1 = [9.0,8.0,7.0,6.0]

  puts "sum 0 is: #{ sum(array0) }\n"
  puts "sum 1 is: #{ sum(array1) }\n"

  puts "average 0 is: #{ average(array0) }\n"
  puts "average 1 is: #{ average(array1) }\n"
end

main

